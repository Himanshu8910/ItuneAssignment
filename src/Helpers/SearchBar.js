import React from 'react';
import {
    TouchableOpacity,
    StyleSheet,
    TextInput,
    Text,
    View,
} from 'react-native';


export default function SearchBar({
    isLoading,
    style,
    onChangeText,
    onSubmit,
    onPress,
}) {
    return (
        <View style={[styles.container, style]}>
            <TextInput
                autoFocus
                style={styles.textInput}
                onChangeText={onChangeText}
                onSubmitEditing={onSubmit}
                numberOfLines={1}
                autoCorrect={false}
                placeholder={'Search artists, songs, albums'}
            />
            <TouchableOpacity style={styles.searchButton} onPress={onPress}>
                <Text style={styles.searchButtonText}>{'Search'}</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 60,
        alignItems: 'center',
        paddingHorizontal: 20,
        borderColor: '#666666',
        borderWidth: 1,
    },
    textInput: {
        flex: 1,
        fontSize: 20,
    },
    loading: {
        marginLeft: 10,
    },
    searchButton: {
        borderColor: '#666666',
        backgroundColor: '#666666',
        borderRadius: 10,
        borderWidth: StyleSheet.hairlineWidth,
        padding: 10,
    },
    searchButtonText: {
        color: 'white',
    },
});
