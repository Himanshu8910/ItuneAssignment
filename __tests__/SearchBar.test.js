import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '../src/Helpers/SearchBar';

test('renders correctly', () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
