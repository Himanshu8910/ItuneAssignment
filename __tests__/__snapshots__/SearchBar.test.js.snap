// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#666666",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    autoFocus={true}
    numberOfLines={1}
    placeholder="Search artists, songs, albums"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#666666",
        "borderColor": "#666666",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Search
    </Text>
  </View>
</View>
`;
